### DISCUSSION TITLE
Improve Quality Handling with Upgrade Options.

### DISCUSSION THREAD

**Discussion by neo-neo1** [pymedusa/Medusa#9989] [Ideas & Feature requests]
Recently installed Radarr and noticed a simple but brilliant option that is missing from Medusa and would greatly improve it. Including options `Upgrade Allowed` and `Upgrade Until Quality = X` [to quality settings](https://i.imgur.com/tB5eJ86.png) is huge welcome. 

_Use Case:_  Most episodes are first released in WEBDL. Medusa downloads it, I watch it, and move on. If BluRay quality available at time of search then yes prefer to download BluRay over WEBDL. But no need to perpetually search and download a better quality BluRay release for what won't be watched again. However, for few select favorite TV shows then yes I do prefer quality upgrades.

At present users must manually mark individual episodes as `Archived` after each download. Having `Upgrade Allowed` options adds not only automated usability but finer quality granularity.

**Comment by p0psicles** [comment]
As also mentioned in discord by @medariox, I think that the existing options are sufficient for your exampled usecase.
But your also right that medusa doesn't offer the flexibility and power in quality management that sonarr does.

But the thing is. That with the existing quality management system we cover like 95% of all users use cases. And maybe sonarr covers 99% (i don't know).

But changing the quality system would 1. mean a lot of work/time 2. would always annoy some users, as not everyone likes change. 3. it will get worse before it gets better. As you'll always have some bugs at the beginning until enough users have tested the new feature.

So up until now we havent bothered changing this to cover that 4% of use cases.

**Comment by TBoneStaek** [comment]
My request may be a long similar lines. Lately I've been enjoying the community's passion projects of upscaling DVDs to 1080p. It'd be nice to have a profile like this so I know that while this show appears to be 1080p, it is not bluray or even native 1080p. That way, if there is a native bluray/1080p released in the future I could more easily see which shows I have could possibly use an upgrade. Currently due to my naming scheme I run two risks: if i name them 1080p.dvd.upscale then Medusa sees them as 1080pHDTV and could potentially upgrade them to a 1080pWEBL but I'd rather this not be automatic. On the flipside, if I label these upscales as dvd.upscale; I also run the issue of upgrading to HDTV or something.

I know I could do a little more tinkering with the profiles but (for example) a long running show like Family Guy which has some great upscales of old episodes, but currently airs 1080pWEBDLs, it would be very difficult to outline this workflow. I've instead taken to marking my upscales as archived so they do not get accidentally replaced by a 1080pWEBDL. (In this scenario, the upscales are preferred because they are uncensored).

Thanks for this passion project!
